packages/updating-element/src/updating-element.ts

```diff
@@ -236,13 +277,19 @@ export interface Controller {
  */
 const finalized = 'finalized';
 
+export type Warnings = 'change-in-update' | 'migration';
+
 /**
  * Base element class which manages element properties and attributes. When
  * properties change, the `update` method is asynchronously called. This method
  * should be supplied by subclassers to render updates as desired.
  * @noInheritDoc
  */
 export abstract class UpdatingElement extends HTMLElement {
+  // Note, these are patched in only in DEV_MODE.
+  static enabledWarnings?: Warnings[];
+  static enableWarning: (type: Warnings) => void;
+  static disableWarning: (type: Warnings) => void;
```

Comment:
I still think it'd be better if these APIs didn't exist at all in prod. We're paying for two fields in prod for here. Will continue discussion below.

Reply:
They do not exist in prod.